### PR TITLE
Fix cusp correction bug

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Breakage since #4053. No release got affected.
The modified unit test fails on current develop but pass with this PR.
I also added a bit more safety guard

## What type(s) of changes does this code introduce?
- Bugfix

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: src/QMCWaveFunctions/LCAO/SoaCuspCorrection.cpp [hunk 1]
```diff
@@ -111,25 +104,22 @@ void SoaCuspCorrection::evaluate_vgl(const ParticleSet& P,
...
       LOBasisSet[c]->evaluate_vgl(dist[c], displ[c], myVGL[0], myVGL[1], myVGL[2], myVGL[3], myVGL[4]);
-    }
-  }
 
   const auto v_in  = myVGL[0];
   const auto gx_in = myVGL[1];
   const auto gy_in = myVGL[2];
   const auto gz_in = myVGL[3];
   const auto l_in  = myVGL[4];
-  for (size_t i = 0; i < psi.rows(); ++i)
```
Comment (ye-luo) [inline src/QMCWaveFunctions/LCAO/SoaCuspCorrection.cpp:132]: This is the line of bug.

File: src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp [hunk 2]
```diff
@@ -528,9 +528,9 @@ TEST_CASE("Ethanol MO with cusp", "[wavefunction]")
   SPOSet::ValueMatrix all_values;
   SPOSet::GradMatrix all_grad;
   SPOSet::ValueMatrix all_lap;
-  all_values.resize(13, 13);
-  all_grad.resize(13, 13);
-  all_lap.resize(13, 13);
+  all_values.resize(10, 13);
```
Comment (markdewing) [inline src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp:531]: Where does the "10" come from?
The following call to "evaluate_notranspose" uses 0,7 for first,last.  For ethanol, that should be 0,14. I think the code was copied from the HCN test and those values were not updated to match.  However, it could be left as-is to test a case where there are fewer electrons than orbitals.

Comment (ye-luo) [reply src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp:531]: You can consider it as a non-ground state calculation with only 10 electron in one spin.

Comment (ye-luo) [reply src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp:531]: From evaluate_notranspose API with first, last, there is no restriction requiring the container size to match exactly. So 0.7 can be considered as a corner case test.

Comment (ye-luo) [reply src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp:531]: @markdewing looks good?

**Review by prckent — APPROVED** [review]
Assuming @markdewing gives assent.

**Review by markdewing — APPROVED** [review]

File: src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp (hunk 2 above)
Comment (markdewing) [reply src/QMCWaveFunctions/tests/test_soa_cusp_corr.cpp:531]: Yes, keeping these values to test non-standard cases is good.

**Comment by ye-luo** [comment]
Test this please
